Mathlib/Probability/Hoeffding.lean

```diff
@@ -0,0 +1,556 @@
+* `ProbabilityTheory.hoeffding`: Hoeffding's Lemma states that for a random variable `X` with
+  `E[X] = 0` (zero mean) and `a ≤ X ≤ b` almost surely, the inequality
+  `μ[exp (t * (X ω))] ≤ exp (t^2 * (b - a)^2 / 8)` holds almost surely for all `t ∈ ℝ`.
+
+## References
+
+* Martin J. Wainwright. High-Dimensional Statistics. Cambridge university press, 2019. Chapter 2
+* Mehryar Mohri. Foundations of Machine Learning. The MIT Press, 2018. Chapter D
+
+## Tags
+
+hoeffding's lemma, cumulant
+-/
+
+open MeasureTheory ProbabilityTheory Real
+
+namespace ProbabilityTheory
+
+universe u
+
+variable {Ω : Type u} [MeasurableSpace Ω] (μ : Measure Ω := by volume_tac)
+
+lemma measurable_expt (X : Ω → ℝ) (t : ℝ) (hX : AEMeasurable X μ) :
+    AEStronglyMeasurable (fun ω ↦ rexp (t * (X ω))) μ :=
+  aestronglyMeasurable_iff_aemeasurable.mpr <| measurable_exp.comp_aemeasurable' (hX.const_mul t)
+
+lemma integrable_expt [IsFiniteMeasure μ] (X : Ω → ℝ) (t b : ℝ) (ht : t > 0)
+    (hX : AEMeasurable X μ) (hb : ∀ᵐ ω ∂μ, X ω ≤ b) :
+    Integrable (fun ω ↦ exp (t * (X ω))) μ := by
+  have hm1 : HasFiniteIntegral (fun ω ↦ rexp (t * X ω)) μ := by
+    have b' : ∀ᵐ ω ∂μ, rexp (t * X ω) ≤ rexp (t * b) := by
+      filter_upwards [hb] with ω hb
+      using exp_le_exp.mpr (mul_le_mul_of_nonneg_left hb (le_of_lt ht))
+    have p : ∀ᵐ ω ∂μ, ‖rexp (t * X ω)‖₊ ≤ rexp (t * b) := by
+      filter_upwards [b'] with ω b'
+      rw [(by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * X ω)‖₊ = rexp (t * X ω))]
+      exact b'
+    have p'' : ∫⁻ ω, ‖rexp (t * X ω)‖₊ ∂μ ≤ ∫⁻ _, ‖rexp (t * b)‖₊ ∂μ := by
+      apply lintegral_mono_ae
+      filter_upwards [p] with ω p
+      simp only [ENNReal.coe_le_coe]
+      rw [← (by simp only [coe_nnnorm, norm_eq_abs, abs_exp] : ‖rexp (t * b)‖₊ = rexp (t * b))] at p
+      exact p
+    suffices ∫⁻ _, ↑‖rexp (t * b)‖₊ ∂μ < ⊤ from lt_of_le_of_lt p'' this
+    simp only [lintegral_const]
+    apply ENNReal.mul_lt_top ENNReal.coe_lt_top (IsFiniteMeasure.measure_univ_lt_top)
+  exact ⟨aestronglyMeasurable_iff_aemeasurable.mpr <|
+    measurable_exp.comp_aemeasurable' (hX.const_mul t), hm1⟩
+
+lemma integrable_expt_bound[IsFiniteMeasure μ] {X : Ω → ℝ} {t a b : ℝ} (hX : AEMeasurable X μ)
+    (h : ∀ᵐ ω ∂μ, X ω ∈ Set.Icc a b) :
+    Integrable (fun ω ↦ exp (t * (X ω))) μ := by
+  have ha : ∀ᵐ ω ∂μ, a ≤ X ω := h.mono fun ω h ↦ h.1
+  cases lt_trichotomy t 0 with
+  | inr ht => cases ht with
+    | inr ht => exact integrable_expt μ X t b ht hX (h.mono fun ω h ↦ h.2)
+    | inl ht => rw [ht]; simp only [zero_mul, exp_zero, integrable_const]
+  | inl ht =>
+    rw [(by ext ω; rw [(by ring : - t * (- X ω) = t * X ω)] :
+      (fun ω ↦ rexp (t * X ω)) = (fun ω ↦ rexp (- t * (- X ω))))]
+    exact integrable_expt _ _ _ _ (by linarith : -t > 0) (AEMeasurable.neg hX)
+      (by filter_upwards [ha] with ω ha using neg_le_neg ha)
+
+lemma expt_pos [IsFiniteMeasure μ] [NeZero μ] (X : Ω → ℝ) (t a b : ℝ)
```

Comment:
We already have [ProbabilityTheory.mgf_pos'](https://leanprover-community.github.io/mathlib4_docs/Mathlib/Probability/Moments.html#ProbabilityTheory.mgf_pos')